docs/internal/release-procedure.md

```diff
@@ -174,3 +174,156 @@ git push origin version-<YOUR_VERSION>
+  ```
+
+- Build:
+
+  ```sh
+  export IS_RELEASE=true \
+  && export CI=true \
+  && ./scripts/package.sh
+  ```
+
+- Artifacts:
+  - You have to upload the following artifacts from `./arduino-ide/electron-app/dist` to S3:
+    - `arduino-ide_${VERSION}_Linux_64bit.AppImage`,
+    - `arduino-ide_${VERSION}_Linux_64bit.zip`, and
+    - `stable-linux.yml`
+
+### macOS M1
+
+- Prerequisites:
+
+  - You need access to the shared Mac Mini at Toolbox. Use TeamViewer.
+  - You have access to the `FT Web Tooling` 1Password vault.
+
+- Clone:
+
+  ```sh
+  BRANCH="<TODO: release branch name>"
+  git clone --single-branch --branch $BRANCH https://github.com/arduino/arduino-ide.git
+  ```
+
+- Setup:
+
+  - Download the `Pro-IDE-Certificates.p12` file from the `Arduino Pro IDE Apple Developer ID Certificate .p12 format` 1Password vault item and put it somewhere on the Mac Mini. This example assumes you put the `.p12` file in the `arduino-ide` repository root, and your `cwd` is also in the `arduino-ide` repository root.
+  - The following environment variables must be available from the shell:
+
+    - `AC_PASSWORD`: From the `Arduino Apple developer ID App Specific Password` 1Password vault item
+    - `AC_USERNAME`: From the `Arduino Apple developer ID App Specific Password` 1Password vault item
+    - `AC_TEAM_ID`: "KT7ZWMCJT"
+    - `CSC_KEY_PASSWORD`: from the `Arduino Pro IDE Apple Developer ID certificate keychain password` 1Password vault item
+    - `CSC_LINK`: Path to the `Pro-IDE-Certificates.p12` file on the Mac Mini
+    - `IS_RELEASE`: "true"
+    - `CAN_SIGN`: "true"
+    - `MACOS_FORCE_NOTARIZE`: "true"
+
+    ```sh
+    export AC_PASSWORD="***"
+    export AC_USERNAME="***"
+    export AC_TEAM_ID="7KT7ZWMCJT"
+    export CSC_KEY_PASSWORD="***"
+    export CSC_LINK="`pwd`/Pro-IDE-Certificates.p12"
+    export IS_RELEASE="true"
+    export CAN_SIGN="true"
+    export MACOS_FORCE_NOTARIZE="true"
+    export CI="true"
+    ```
+
+- Build:
+
+  ```sh
+  ./scripts/package.sh
+  ```
+
+  > Make sure to change directory, and execute the script from the root of the `arduino-ide` repository.
+
+- Verify:
+
+  Since you cannot drag and drop via TeamViewer, you will install the app from a command line. This example puts the IDE2 into the Desktop. Do **NOT** try to bypass the installation with a double click and open from the UI. The `node_modules` folder of the source code is implicitly in the `$PATH`, and you want to verify if the app is fully functional without the `node_modules` folder.
+
+  ```sh
+  VERSION="<TODO: release version>"
+  hdiutil attach ./electron-app/dist/arduino-ide_${VERSION}_macOS_arm64.dmg \
+  && cp -R /Volumes/Arduino\ IDE\ ${VERSION}-arm64/Arduino\ IDE.app ~/Desktop \
+  && hdiutil unmount /Volumes/Arduino\ IDE\ ${VERSION}-arm64 \
+  && codesign -dv --verbose=4 ~/Desktop/Arduino\ IDE.app \
+  && ~/Desktop/Arduino\ IDE.app/Contents/MacOS/Arduino\ IDE
+  ```
+
```

Comment:
Should manually rename `arm` to `ARM` before the upload: https://github.com/arduino/arduino-ide/issues/2117